From 34e2a652a63e5dc97b38b19bd83301f7bfddbaaa Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Tue, 8 Feb 2022 17:24:42 +0200
Subject: [PATCH 35/43] dts: Remove unused clk modules nodes

Removed all the clocking modules nodes from the s32-gen1
platforms device-trees:
 - armpll, armdfs, periphpll, periphdfs, accelpll, ddrpll,
mc_me, rdc, rgm, mc_cgm0, mc_cgm1, mc_cgm2, mc_cgm5

Issue: ALB-8352
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/fsl-s32-gen1.dtsi | 192 ---------------------------------
 arch/arm/dts/fsl-s32g.dtsi     | 114 --------------------
 arch/arm/dts/fsl-s32g3.dtsi    |  78 --------------
 arch/arm/dts/fsl-s32r45.dts    |  68 ------------
 4 files changed, 452 deletions(-)

diff --git a/arch/arm/dts/fsl-s32-gen1.dtsi b/arch/arm/dts/fsl-s32-gen1.dtsi
index 1fc6af34f9..bbcde81de9 100644
--- a/arch/arm/dts/fsl-s32-gen1.dtsi
+++ b/arch/arm/dts/fsl-s32-gen1.dtsi
@@ -230,198 +230,6 @@
 		status = "disabled";
 	};
 
-	clks {
-		compatible = "fsl,s32-gen1-clocking";
-		#address-cells = <2>;
-		#size-cells = <2>;
-		#clock-cells = <1>;
-		status = "disabled";
-
-		assigned-clocks =
-			<&clks S32GEN1_SCMI_CLK_SERDES_REF>,
-			<&clks S32GEN1_CLK_FXOSC>,
-			<&clks S32GEN1_CLK_FIRC>,
-			<&clks S32GEN1_CLK_SIRC>,
-			<&clks S32GEN1_CLK_GMAC0_EXT_RX>;
-
-		assigned-clock-parents =
-			<0>,
-			<&fxosc 0>,
-			<&firc 0>,
-			<&sirc 0>,
-			<&gmac0_ext_rx 0>;
-
-		assigned-clock-rates =
-			<100000000>;
-
-		armpll: armpll@4003800 {
-			compatible = "fsl,s32gen1-armpll";
-			reg = <0x0 0x40038000 0x0 0x3000>;
-
-			assigned-clocks =
-				<&clks S32GEN1_CLK_ARM_PLL_MUX>,
-				<&clks S32GEN1_CLK_ARM_PLL_VCO>,
-				<&clks S32GEN1_CLK_ARM_PLL_PHI0>;
-			assigned-clock-parents =
-				<&clks S32GEN1_CLK_FXOSC>;
-		};
-
-		armdfs: armdfs@40054000 {
-			compatible = "fsl,s32gen1-armdfs";
-			reg = <0x0 0x40054000 0x0 0x3000>;
-
-			assigned-clocks =
-				<&clks S32GEN1_CLK_ARM_PLL_DFS1>,
-				<&clks S32GEN1_CLK_ARM_PLL_DFS2>;
-			assigned-clock-rates =
-				<0>,
-				<800000000>;
-		};
-
-		periphpll: periphpll@4003C000 {
-			compatible = "fsl,s32gen1-periphpll";
-			reg = <0x0 0x4003C000 0x0 0x3000>;
-
-			assigned-clocks =
-				<&clks S32GEN1_CLK_PERIPH_PLL_MUX>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_VCO>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI0>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI2>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI3>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI4>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI5>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI7>;
-			assigned-clock-parents =
-				<&clks S32GEN1_CLK_FXOSC>;
-			assigned-clock-rates =
-				<0>,
-				<S32GEN1_PERIPH_PLL_VCO_FREQ>, <100000000>,
-				<80000000>, <80000000>,
-				<125000000>, <200000000>,
-				<125000000>, <100000000>;
-		};
-
-		periphdfs: armdfs@40058000 {
-			compatible = "fsl,s32gen1-periphdfs";
-			reg = <0x0 0x40058000 0x0 0x3000>;
-
-			assigned-clocks =
-				<&clks S32GEN1_CLK_PERIPH_PLL_DFS1>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_DFS3>;
-			assigned-clock-rates =
-				<800000000>,
-				<800000000>;
-		};
-
-		accelpll: accelpll@40040000 {
-			compatible = "fsl,s32gen1-accelpll";
-			reg = <0x0 0x40040000 0x0 0x3000>;
-		};
-
-		ddrpll: ddrpll@40044000 {
-			compatible = "fsl,s32gen1-ddrpll";
-			reg = <0x0 0x40044000 0x0 0x3000>;
-		};
-
-		mc_me: mc_me@40088000 {
-			compatible = "fsl,s32gen1-mc_me";
-			reg = <0x0 0x40088000 0x0 0x3000>;
-		};
-
-		rdc: rdc@440080000 {
-			compatible = "fsl,s32gen1-rdc";
-			reg = <0x0 0x40080000 0x0 0x3000>;
-		};
-
-		rgm: rgm@40078000 {
-			compatible = "fsl,s32gen1-rgm";
-			reg = <0x0 0x40078000 0x0 0x3000>;
-		};
-
-		mc_cgm0: mc_cgm0@40030000 {
-			compatible = "fsl,s32gen1-mc_cgm0";
-			reg = <0x0 0x40030000 0x0 0x3000>;
-
-			assigned-clocks =
-				<&clks S32GEN1_CLK_MC_CGM0_MUX0>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX3>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX4>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX5>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX7>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX8>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX9>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX10>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX11>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX12>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX14>,
-				<&clks S32GEN1_CLK_MC_CGM0_MUX16>,
-				<&clks S32GEN1_CLK_PER>,
-				<&clks S32GEN1_CLK_XBAR_2X>,
-				<&clks S32GEN1_CLK_FTM0_REF>,
-				<&clks S32GEN1_CLK_FTM1_REF>,
-				<&clks S32GEN1_CLK_CAN_PE>,
-				<&clks S32GEN1_CLK_LIN_BAUD>,
-				<&clks S32GEN1_CLK_SPI>,
-				<&clks S32GEN1_CLK_QSPI_2X>,
-				<&clks S32GEN1_CLK_SDHC>,
-				<&clks S32GEN1_CLK_GMAC0_TS>;
-			assigned-clock-parents =
-				<&clks S32GEN1_CLK_ARM_PLL_DFS1>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI2>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI3>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI4>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI5>,
-				<&clks S32GEN1_CLK_GMAC0_EXT_RX>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_DFS1>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_DFS3>,
-				<&clks S32GEN1_CLK_PERIPH_PLL_PHI7>;
-			assigned-clock-rates =
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<0>,
-				<80000000>,
-				<0>,
-				<40000000>,
-				<40000000>,
-				<80000000>,
-				<125000000>,
-				<100000000>,
-				<400000000>,
-				<400000000>,
-				<200000000>;
-		};
-
-		mc_cgm1: mc_cgm1@40034000 {
-			compatible = "fsl,s32gen1-mc_cgm1";
-			reg = <0x0 0x40034000 0x0 0x3000>;
-
-			assigned-clocks =
-				<&clks S32GEN1_CLK_MC_CGM1_MUX0>,
-				<&clks S32GEN1_CLK_A53_CORE>;
-			assigned-clock-parents =
-				<&clks S32GEN1_CLK_ARM_PLL_PHI0>,
-				<0>;
-		};
-
-		mc_cgm5: mc_cgm5@40068000 {
-			compatible = "fsl,s32gen1-mc_cgm5";
-			reg = <0x0 0x40068000 0x0 0x3000>;
-		};
-	};
-
 	uart0:serial@401C8000 {
 		device_type = "serial";
 		compatible = "fsl,s32-linflexuart";
diff --git a/arch/arm/dts/fsl-s32g.dtsi b/arch/arm/dts/fsl-s32g.dtsi
index bf9b447f98..b2cc9a30e0 100644
--- a/arch/arm/dts/fsl-s32g.dtsi
+++ b/arch/arm/dts/fsl-s32g.dtsi
@@ -51,18 +51,6 @@
 		device_id = <1>;
 		num-lanes = <2>; /* supports max 2 lanes */
 		clocks = <&clks S32GEN1_SCMI_CLK_SERDES_REF>;
-
-		assigned-clocks =
-			<&clks S32G_CLK_SERDES1_LANE0_TX>,
-			<&clks S32G_CLK_SERDES1_LANE0_CDR>,
-			<&clks S32G_CLK_SERDES1_LANE1_TX>,
-			<&clks S32G_CLK_SERDES1_LANE1_CDR>;
-		assigned-clock-parents =
-			<&serdes1_lane0_ext_tx 0>,
-			<&serdes1_lane0_ext_cdr 0>,
-			<&serdes1_lane1_ext_tx 0>,
-			<&serdes1_lane1_ext_cdr 0>;
-
 		status = "okay";
 	};
 
@@ -256,96 +244,6 @@
 	};
 };
 
-&clks {
-	clocks = <&clks S32GEN1_SCMI_CLK_FTM0_SYS>,
-		<&clks S32GEN1_SCMI_CLK_FTM1_SYS>,
-		<&clks S32GEN1_SCMI_CLK_FTM0_EXT>,
-		<&clks S32GEN1_SCMI_CLK_FTM1_EXT>,
-		<&clks S32GEN1_SCMI_CLK_FLEXCAN_CAN>,
-		<&clks S32GEN1_SCMI_CLK_SPI_REG>,
-		<&clks S32GEN1_SCMI_CLK_QSPI_FLASH1X>;
-
-	mc_cgm2: mc_cgm2@44018000 {
-		compatible = "fsl,s32gen1-mc_cgm2";
-		reg = <0x0 0x44018000 0x0 0x3000>;
-
-		assigned-clocks =
-			<&clks S32G_CLK_PFE_MAC0_EXT_REF>,
-			<&clks S32G_CLK_PFE_MAC1_EXT_REF>,
-			<&clks S32G_CLK_PFE_MAC2_EXT_REF>,
-			<&clks S32G_CLK_MC_CGM2_MUX0>,
-			/* PFE MAC0 */
-			<&clks S32G_CLK_MC_CGM2_MUX7>,
-			<&clks S32G_CLK_MC_CGM2_MUX4>,
-			<&clks S32G_CLK_MC_CGM2_MUX1>,
-			/* PFE MAC1 */
-			<&clks S32G_CLK_MC_CGM2_MUX8>,
-			<&clks S32G_CLK_MC_CGM2_MUX5>,
-			<&clks S32G_CLK_MC_CGM2_MUX2>,
-			/* PFE MAC2 */
-			<&clks S32G_CLK_MC_CGM2_MUX9>,
-			<&clks S32G_CLK_MC_CGM2_MUX6>,
-			<&clks S32G_CLK_MC_CGM2_MUX3>,
-
-			<&clks S32G_CLK_PFE_PE>,
-			<&clks S32G_CLK_PFE_MAC0_REF_DIV>,
-			<&clks S32G_CLK_PFE_MAC0_RX>,
-			<&clks S32G_CLK_PFE_MAC0_TX_DIV>,
-			<&clks S32G_CLK_PFE_MAC1_REF_DIV>,
-			<&clks S32G_CLK_PFE_MAC1_RX>,
-			<&clks S32G_CLK_PFE_MAC1_TX>,
-			<&clks S32G_CLK_PFE_MAC2_REF_DIV>,
-			<&clks S32G_CLK_PFE_MAC2_RX>,
-			<&clks S32G_CLK_PFE_MAC2_TX>;
-		assigned-clock-parents =
-			<&pfe_mac0_ext_ref 0>,
-			<&pfe_mac1_ext_ref 0>,
-			<&pfe_mac2_ext_ref 0>,
-			<&clks S32G_CLK_ACCEL_PLL_PHI1>,
-			/* PFE MAC0 */
-			<&clks S32G_CLK_PFE_MAC0_EXT_REF>,
-			<&clks S32G_CLK_SERDES1_LANE0_CDR>,
-			<&clks S32G_CLK_SERDES1_LANE0_TX>,
-			/* PFE MAC1 */
-			<&clks S32G_CLK_PFE_MAC1_EXT_REF>,
-			<&clks S32G_CLK_SERDES1_LANE1_CDR>,
-			<&clks S32G_CLK_SERDES1_LANE1_TX>,
-			/* PFE MAC2 */
-			<&clks S32G_CLK_PFE_MAC2_EXT_REF>,
-			<&clks S32G_CLK_SERDES0_LANE1_CDR>,
-			<&clks S32G_CLK_SERDES0_LANE1_TX>;
-		assigned-clock-rates =
-			<0>, <0>, <0>, <0>, <0>, <0>, <0>,
-			<0>, <0>, <0>, <0>, <0>, <0>,
-			<600000000>,
-			<50000000>,
-			<125000000>,
-			<125000000>,
-			<50000000>,
-			<125000000>,
-			<125000000>,
-			<50000000>,
-			<125000000>,
-			<125000000>;
-
-	};
-};
-
-&accelpll {
-	assigned-clocks =
-		<&clks S32GEN1_CLK_ACCEL_PLL_MUX>,
-		<&clks S32GEN1_CLK_ACCEL_PLL_VCO>,
-		<&clks S32G_CLK_ACCEL_PLL_PHI0>,
-		<&clks S32G_CLK_ACCEL_PLL_PHI1>;
-	assigned-clock-parents =
-		<&clks S32GEN1_CLK_FXOSC>;
-	assigned-clock-rates =
-		<0>,
-		<1800000000>,
-		<600000000>,
-		<600000000>;
-};
-
 &gmac0 {
 	clocks = <&clks S32GEN1_SCMI_CLK_GMAC0_TS>,
 		 <&clks S32GEN1_SCMI_CLK_GMAC0_RX_SGMII>,
@@ -391,18 +289,6 @@
 };
 
 &serdes0 {
-	assigned-clocks =
-		<&clks S32GEN1_CLK_SERDES0_LANE0_TX>,
-		<&clks S32GEN1_CLK_SERDES0_LANE0_CDR>,
-		<&clks S32G_CLK_SERDES0_LANE1_TX>,
-		<&clks S32G_CLK_SERDES0_LANE1_CDR>;
-
-	assigned-clock-parents =
-		<&serdes0_lane0_ext_tx 0>,
-		<&serdes0_lane0_ext_cdr 0>,
-		<&serdes0_lane1_ext_tx 0>,
-		<&serdes0_lane1_ext_cdr 0>;
-
 	status = "okay";
 };
 
diff --git a/arch/arm/dts/fsl-s32g3.dtsi b/arch/arm/dts/fsl-s32g3.dtsi
index c46df35f02..0d6394e325 100644
--- a/arch/arm/dts/fsl-s32g3.dtsi
+++ b/arch/arm/dts/fsl-s32g3.dtsi
@@ -40,8 +40,6 @@
 	};
 };
 
-/delete-node/ &mc_cgm0;
-
 &cpus {
 	cpu4: cpu@2 {
 		device_type = "cpu";
@@ -68,82 +66,6 @@
 	};
 };
 
-&clks {
-	mc_cgm0: mc_cgm0@40030000 {
-		compatible = "fsl,s32gen1-mc_cgm0";
-		reg = <0x0 0x40030000 0x0 0x3000>;
-
-		assigned-clocks =
-			<&clks S32GEN1_CLK_MC_CGM0_MUX0>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX3>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX4>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX5>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX7>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX8>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX12>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX14>,
-			<&clks S32GEN1_CLK_MC_CGM0_MUX16>,
-			<&clks S32GEN1_CLK_PER>,
-			<&clks S32GEN1_CLK_XBAR_2X>,
-			<&clks S32GEN1_CLK_FTM0_REF>,
-			<&clks S32GEN1_CLK_FTM1_REF>,
-			<&clks S32GEN1_CLK_CAN_PE>,
-			<&clks S32GEN1_CLK_LIN_BAUD>,
-			<&clks S32GEN1_CLK_SPI>,
-			<&clks S32GEN1_CLK_QSPI_2X>,
-			<&clks S32GEN1_CLK_SDHC>;
-		assigned-clock-parents =
-			<&clks S32GEN1_CLK_ARM_PLL_DFS1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI2>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI3>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_DFS1>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_DFS3>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI7>;
-		assigned-clock-rates =
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<0>,
-			<80000000>,
-			<0>,
-			<40000000>,
-			<40000000>,
-			<80000000>,
-			<125000000>,
-			<100000000>,
-			<400000000>,
-			<400000000>;
-	};
-
-	mc_cgm6: mc_cgm0@4053C000 {
-		compatible = "fsl,s32gen1-mc_cgm6";
-		reg = <0x0 0x4053C000 0x0 0x3000>;
-
-		assigned-clocks =
-			<&clks S32G_CLK_MC_CGM6_MUX0>,
-			<&clks S32G_CLK_MC_CGM6_MUX1>,
-			<&clks S32G_CLK_MC_CGM6_MUX2>,
-			<&clks S32GEN1_CLK_GMAC0_TS>;
-		assigned-clock-parents =
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI4>,
-			<&clks S32GEN1_CLK_PERIPH_PLL_PHI5>,
-			<&clks S32GEN1_CLK_GMAC0_EXT_RX>;
-		assigned-clock-rates =
-			<0>,
-			<0>,
-			<0>,
-			<200000000>;
-	};
-};
-
 &siul2_1_nvram {
 	compatible = "fsl,s32g3-siul2_1-nvram";
 };
diff --git a/arch/arm/dts/fsl-s32r45.dts b/arch/arm/dts/fsl-s32r45.dts
index 62f59d23f5..eafb0a70e5 100644
--- a/arch/arm/dts/fsl-s32r45.dts
+++ b/arch/arm/dts/fsl-s32r45.dts
@@ -132,15 +132,6 @@
 		device_id = <1>;
 		num-lanes = <1>; /* supports 1 lane */
 		clocks = <&clks S32GEN1_SCMI_CLK_SERDES_REF>;
-
-		assigned-clocks =
-			<&clks S32R45_CLK_SERDES1_LANE0_TX>,
-			<&clks S32R45_CLK_SERDES1_LANE0_CDR>;
-
-		assigned-clock-parents =
-			<&serdes1_lane0_ext_tx 0>,
-			<&serdes1_lane0_ext_cdr 0>;
-
 		status = "okay";
 	};
 
@@ -196,65 +187,6 @@
 	};
 };
 
-&serdes0 {
-	assigned-clocks =
-		<&clks S32GEN1_CLK_SERDES0_LANE0_TX>,
-		<&clks S32GEN1_CLK_SERDES0_LANE0_CDR>,
-		<&clks S32GEN1_CLK_SERDES_REF>;
-	assigned-clock-parents =
-		<&serdes0_lane0_ext_tx 0>,
-		<&serdes0_lane0_ext_cdr 0>;
-	assigned-clock-rates =
-		<0>, <0>,
-		<100000000>;
-};
-
-&accelpll {
-	assigned-clocks =
-		<&clks S32GEN1_CLK_ACCEL_PLL_MUX>,
-		<&clks S32GEN1_CLK_ACCEL_PLL_VCO>,
-		<&clks S32R45_CLK_ACCEL_PLL_PHI0>,
-		<&clks S32GEN1_CLK_ARM_PLL_DFS4>;
-	assigned-clock-parents =
-		<&clks S32GEN1_CLK_FXOSC>;
-	assigned-clock-rates =
-		<0>,
-		<1800000000>,
-		<600000000>,
-		<400000000>;
-};
-
-&clks {
-	clocks = <&clks S32GEN1_SCMI_CLK_FTM0_SYS>,
-		<&clks S32GEN1_SCMI_CLK_FTM1_SYS>,
-		<&clks S32GEN1_SCMI_CLK_FTM0_EXT>,
-		<&clks S32GEN1_SCMI_CLK_FTM1_EXT>,
-		<&clks S32GEN1_SCMI_CLK_FLEXCAN_CAN>,
-		<&clks S32GEN1_SCMI_CLK_SPI_REG>,
-		<&clks S32GEN1_SCMI_CLK_QSPI_FLASH1X>,
-		<&clks S32R45_SCMI_CLK_LAX_MODULE>,
-		<&clks S32R45_SCMI_CLK_SPT_SPT>;
-
-	mc_cgm2: mc_cgm2@440C0000 {
-		compatible = "fsl,s32gen1-mc_cgm2";
-		reg = <0x0 0x440C0000 0x0 0x3000>;
-
-		assigned-clocks =
-			<&clks S32R45_CLK_MC_CGM2_MUX0>,
-			<&clks S32R45_CLK_MC_CGM2_MUX1>,
-			<&clks S32R45_CLK_ACCEL3>,
-			<&clks S32R45_CLK_ACCEL4>;
-		assigned-clock-parents =
-			<&clks S32R45_CLK_ACCEL_PLL_PHI0>,
-			<&clks S32R45_CLK_ARM_PLL_DFS4_2>;
-		assigned-clock-rates =
-			<0>,
-			<0>,
-			<600000000>,
-			<400000000>;
-	};
-};
-
 &qspi {
 	pinctrl-0 = <&pinctrl0_qspi>;
 	pinctrl-names = "default";
-- 
2.17.1

